src/service/extensions-impl.js

```diff
@@ -452,27 +454,29 @@ export class Extensions {
 /**
  * Calculate script url for amp-ad.
  * @visibleForTesting
+ * @param {string} protocol Serving protocol
  * @param {string} path Location path of the window
+ * @param {string} port Serving port
  * @param {string} extensionId
  * @param {boolean=} isTest
  * @param {boolean=} isUsingCompiledJs
  * @return {string}
  * @visibleForTesting
  */
-export function calculateExtensionScriptUrl(path, extensionId, isTest,
-    isUsingCompiledJs) {
+export function calculateExtensionScriptUrl(protocol, path, port, extensionId,
+    isTest, isUsingCompiledJs) {
   if (getMode().localDev) {
     if (isTest) {
       if (isUsingCompiledJs) {
         return `/base/dist/v0/${extensionId}-0.1.js`;
       }
       return `/base/dist/v0/${extensionId}-0.1.max.js`;
     }
-    if (path.indexOf('.max') >= 0) {
-      return `http://localhost:8000/dist/v0/${extensionId}-0.1.max.js`;
+    if (path.indexOf('.max') >= 0 || path.substr(0, 5) == '/max/') {
+      return `${protocol}//localhost:${port}/dist/v0/${extensionId}-0.1.max.js`;
```

Comment:
The only cases I know of are for `ads.localhost`, but in that case it's essentially a `CNAME` so it'll serve up the docs without problem.